src/commands/Management/Configuration/manageCommandChannel.ts

```diff
@@ -23,101 +24,103 @@ export default class extends SkyraCommand {
+		const [disabledCommandsChannels, language] = await message.guild.readSettings((settings) => [
+			settings[GuildSettings.DisabledCommandChannels],
+			settings.getLanguage()
+		]);
+
 		const entry = disabledCommandsChannels.find((e) => e.channel === channel.id);
 		if (entry && entry.commands.length) {
-			return message.sendLocale(LanguageKeys.Commands.Management.ManageCommandChannelShow, [
-				{ channel: channel.toString(), commands: `\`${entry.commands.join('` | `')}\`` }
-			]);
+			return message.send(
+				language.get(LanguageKeys.Commands.Management.ManageCommandChannelShow, {
+					channel: channel.toString(),
+					commands: `\`${entry.commands.join('` | `')}\``
+				})
+			);
 		}
-		throw message.language.get(LanguageKeys.Commands.Management.ManageCommandChannelShowEmpty);
+
+		throw language.get(LanguageKeys.Commands.Management.ManageCommandChannelShowEmpty);
 	}
 
-	public async add(message: KlasaMessage, [channel, command]: [TextChannel, SkyraCommand]) {
-		const disabledCommandsChannels = message.guild!.settings.get(GuildSettings.DisabledCommandChannels);
-		const index = disabledCommandsChannels.findIndex((e) => e.channel === channel.id);
-
-		if (index === -1) {
-			await message.guild!.settings.update(
-				GuildSettings.DisabledCommandChannels,
-				{ channel: channel.id, commands: [command.name] },
-				{
-					arrayAction: 'add',
-					extraContext: { author: message.author.id }
-				}
-			);
-		} else {
-			const entry = disabledCommandsChannels[index];
-			if (entry.commands.includes(command.name)) throw message.language.get(LanguageKeys.Commands.Management.ManageCommandChannelAddAlreadyset);
-
-			await message.guild!.settings.update(
-				GuildSettings.DisabledCommandChannels,
-				{ channel: entry.channel, commands: entry.commands.concat(command.name) },
-				{
-					arrayIndex: index,
-					extraContext: { author: message.author.id }
-				}
-			);
-		}
-		return message.sendLocale(LanguageKeys.Commands.Management.ManageCommandChannelAdd, [{ channel: channel.toString(), command: command.name }]);
+	public async add(message: GuildMessage, [channel, command]: [TextChannel, SkyraCommand]) {
+		const language = await message.guild.writeSettings((settings) => {
+			const disabledCommandsChannels = settings[GuildSettings.DisabledCommandChannels];
+			const indexOfChannel = disabledCommandsChannels.findIndex((e) => e.channel === channel.id);
+			const language = settings.getLanguage();
+
+			if (indexOfChannel === -1) {
+				settings[GuildSettings.DisabledCommandChannels].push({ channel: channel.id, commands: [command.name] });
+			} else {
+				const disabledCommandChannel = disabledCommandsChannels[indexOfChannel];
+				if (disabledCommandChannel.commands.includes(command.name))
+					throw language.get(LanguageKeys.Commands.Management.ManageCommandChannelAddAlreadyset);
+
+				settings[GuildSettings.DisabledCommandChannels][indexOfChannel].commands.concat(command.name);
```

Comment:
This won't do anything; concat returns a new array. Did you mean `push`?
```suggestion
				settings[GuildSettings.DisabledCommandChannels][indexOfChannel].commands.push(command.name);
```